vignettes/miaTime.Rmd

```diff
@@ -95,20 +117,153 @@ Timediff <- c(NA, Timediff)
 Timediff[1140:1151]
 ```
 
-Also, the time difference from selected point to rest of each time point 
-can be calculated.
+The time difference from a selected point to the other time points
+can be calculated as follows.
+
 ```{r}
 base <- Hours - Hours[1] #distance from starting point
 base[1140:1151]
 
 base_1140 <- Seconds - Seconds[1140]
 base_1140[1140:1151]
-
 ```
+
+## Time point rank
+
 Rank of the time points can be calculated by `rank` function provided in base R.
+
+```{r}
+colData(se)$rank <- rank(colData(se)$time)
+colData(se)
+```
+
+
+## Operations per unit
+
+Sometimes we need to operate on time series per unit (subject,
+reaction chamber, sampling location, ...).
+
+Add time point rank per subject.
+
 ```{r}
-colData(hitchip1006)$rank <- rank(colData(hitchip1006)$time)
-colData(hitchip1006)
+library(dplyr)
+colData(se) <- colData(se) %>%
+   as.data.frame() %>%
+   group_by(subject) %>%
+   mutate(rank = rank(time, ties.method="average")) %>%
+   DataFrame()
+```
+
+
+
+Calculate the difference in time from a given baseline time point.
+
+```{r}
+# Define baseline function;
+base <- function (x, baseline=min(x)) {x - baseline}
+
+# Use time 1 as the baseline
+colData(se) <- colData(se) %>%
+    as.data.frame() %>%
+    group_by(subject) %>%
+    mutate(shifted = base(time, baseline=1)) %>%
+    DataFrame()
+# Check output
+as.data.frame(colData(se)[, c("subject", "time", "shifted")]) %>%
+    dplyr::filter(subject == "831")
+
+# Use the minimum time point per subject as the baseline:
+colData(se) <- colData(se) %>%
+    as.data.frame() %>%
+    group_by(subject) %>%
+    mutate(shifted = base(time)) %>%
+    DataFrame()
+
+# Check output
+as.data.frame(colData(se)[, c("subject", "time", "shifted")]) %>%
+    dplyr::filter(subject == "831")
+```
+
+
+## Estimating divergence and convergence
+
+[@gutMicrobiota] is available for further details.
```

Comment:
I couldn't access to full article, so I couldn't find the relation and left it like this for now.